server/src/main/java/io/littlehorse/common/model/wfrun/ThreadRun.java

```diff
@@ -885,17 +883,22 @@ public NodeRun getNodeRun(int position) {
         return out;
     }
 
-    public void putVariable(String varName, VariableValue var) throws LHVarSubError {
+    /**
+     * Traverse the current and parent ThreadRun to run the variable modification on the appropriate ThreadRun
+     * @param varName name of the variable
+     * @param function function that executes the variable modification (e.g. creation, mutation, etc.)
+     * @throws LHVarSubError when the varName is not found either on the current ThreadRun definition
+     * or its parents definition
+     */
+    private void applyOnAppropriateThread(
```

Comment:
Is this to complicated or hard to understand? Since we divided the `putVariable` into a `createVariable` and a `mutateVariable`, I wanted a shared way of applying the same parent traversing logic the `putVariable` was applying. If it's to hard to understand and/or maintain I can change it.
cc: @coltmcnealy-lh